src/main/java/nextstep/subway/line/domain/Sections.java

```diff
@@ -0,0 +1,139 @@
+    private static final int MIN_LIMIT = 1;
+
+    @OneToMany(mappedBy = "line", cascade = {CascadeType.PERSIST, CascadeType.MERGE}, orphanRemoval = true)
+    private final List<Section> sections = new ArrayList<>();
+
+    public static Sections create() {
+        return new Sections();
+    }
+
+    public void add(final Section newSection) {
+        if (this.sections.isEmpty()) {
+            sections.add(newSection);
+            return;
+        }
+
+        final List<Station> stations = getStations();
+
+        final boolean isUpStationExisted = isExistStation(station -> station == newSection.getUpStation());
+        final boolean isDownStationExisted = isExistStation(station -> station == newSection.getDownStation());
+
+        validateStations(newSection, stations, isUpStationExisted, isDownStationExisted);
+
+        changeOriginSection(newSection, isUpStationExisted, isDownStationExisted);
+
+        sections.add(newSection);
+    }
+
+    private boolean isExistStation(final Predicate<Station> predicate) {
+        return getStations().stream().anyMatch(predicate);
+    }
+
+    private void validateStations(final Section newSection, final List<Station> stations,
+                                  final boolean isUpStationExisted, final boolean isDownStationExisted) {
+        if (isUpStationExisted && isDownStationExisted) {
+            throw new RuntimeException(ERR_TEXT_ALREADY_ADDED_SECTION);
+        }
+
+        if (!stations.contains(newSection.getUpStation()) && !stations.contains(newSection.getDownStation())) {
+            throw new RuntimeException(ERR_TEXT_CAN_NOT_ADD_SECTION);
+        }
+    }
+
+    private void changeOriginSection(final Section newSection, final boolean isUpStationExisted, final boolean isDownStationExisted) {
+        if (isUpStationExisted) {
+            findSection(section -> section.isMatchUpAndUpStation(newSection))
+                .ifPresent(section -> section.updateUpStationByNewSection(newSection));
+            return;
+        }
+
+        if (isDownStationExisted) {
+            findSection(section -> section.isMatchDownAndDownStation(newSection))
+                .ifPresent(section -> section.updateDownStationByNewSection(newSection));
+        }
+    }
+
+    public void remove(final Station targetStation, final Line line) {
+        isConditionThatCanBeDeleted(targetStation);
+
+        final Optional<Section> upLineStation = findSection(section -> section.isMatchUpStation(targetStation));
+        final Optional<Section> downLineStation = findSection(section -> section.isMatchDownStation(targetStation));
+
+        if (upLineStation.isPresent() && downLineStation.isPresent()) {
+            addSectionWhenRemoveStationsIsIncludedInMoreThanOne(line, upLineStation.get(), downLineStation.get());
+        }
+
+        upLineStation.ifPresent(sections::remove);
+        downLineStation.ifPresent(sections::remove);
```

Comment:
이건 개인적인 의견이지만, `List<Section>`을 활용해도 괜찮을 것 같아요.

라인은 조금 줄어들긴 하지만 up, down이 명확히 적혀있지 않아 조금 헷갈릴수는 있겠네요

참고만 부탁드리겠습니다~

```suggestion
        final List<Section> relatedSections = ();

        if (relatedSections.size == 2) {
            addSectionWhenRemoveStationsIsIncludedInMoreThanOne(line, upLineStation.get(), downLineStation.get());
        }

        relatedSections.forEach(sections::remove);
```

Reply:
피드백 주신 방법도 괜찮은 것 같습니다!
단지 스타일의 차이라 말씀 주신대로 처리해도 간단한 로직이기 때문에 크게 보기 어렵진 않을거라 생각합니다 ㅎㅎ